runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6754,6 +6754,23 @@ TR_J9ByteCodeIlGenerator::genNewArray(int32_t typeIndex)
         }
      }
 
+   // Special case for handling Arrays.copyOfRangeByte when the caller of caller is String<init>(AbstractStringBuilder, Void).
+   // The caller is always Arrays.copyOfRange since Arrays.copyOfRangeByte is a private method.
+   if ((comp()->getInlineDepth() > 1) && !comp()->isPeekingMethod() && !generateArraylets
+       && _methodSymbol->getRecognizedMethod() == TR::java_util_Arrays_copyOfRangeByte)
+      {
+      int32_t callerIndex = comp()->getCurrentInlinedCallSite()->_byteCodeInfo.getCallerIndex();
+      int32_t callerOfCallerIndex = comp()->getInlinedCallSite(callerIndex)._byteCodeInfo.getCallerIndex();
+      TR::ResolvedMethodSymbol *callerOfCaller = callerOfCallerIndex > -1 ? comp()->getInlinedResolvedMethodSymbol(callerOfCallerIndex) : comp()->getOptimizer()->getMethodSymbol();
```

Comment:
When the index is -1, I thought I could use `comp()->getMethodSymbol()` but it returns the symbol for the inner most method (`copyOfRangeByte`) unless `TR_DisableReturnCalleeInIlgen` env var is set. It is funny that when I use  `comp()->getMethodSymbol()` in dynamic debug counters it returns the outer most method symbol (index -1) but when use it here it returns the inner most method symbol! `comp()->getOptimizer()->getMethodSymbol()` seemingly works fine but I need to check with @r30shah

Reply:
This question is still valid